packages/venia-ui/lib/components/FilterModal/filterModal.js

```diff
@@ -66,9 +69,49 @@ const FilterModal = props => {
         </div>
     ) : null;
 
+    function changeFocusOnModal(event) {
+        if (isOpen) {
+            event.target.focus();
+        }
+    }
+
+    function handleKeyActions(event) {
+        if (isOpen) {
+            if (event.keyCode === 27) {
+                handleClose(event);
+            }
+            const element = modalRef.current;
+            const focusableEls = tabbable(element);
+            const firstFocusableEl = focusableEls[0];
+            const lastFocusableEl = focusableEls[focusableEls.length - 1];
```

Comment:
This seems like something you could initialize on open and not need to calculate again on every key press; since we just need to measure once the ref is attached, I'm thinking a callback ref is the best solution here.

```jsx
const filterModalRef = useCallback(node => {
  if (node !== null) {
    // do your calc here and set first/last into local state variable
  }
}, []);
```